cmd/contour/serve.go

```diff
@@ -352,6 +383,99 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	}
 	g.Add(debugsvc.Start)
 
+	// Leader Election starts here.
+
+	leaderOK := make(chan struct{})
+	deposed := make(chan struct{})
+	leadingDisabled := make(chan struct{})
+
+	g.Add(func(stop <-chan struct{}) error {
```

Comment:
I suspect that 392-464 could be executed in the context of the current goroutine, not handed off to the group that would leave
```
		// Running the leader election requires a context.Context, but we're
		// not using them, so we'll just pass in an empty one.
		le.Run(electionCtx)
		return nil
```
to be run in a goroutine. This probably means the context handling needs to be redone. Perhaps we need a better way to integrate contexts with workgroups.